course-v2/assets/js/video_download_popup.ts

```diff
@@ -0,0 +1,34 @@
+export const initVideoDownloadPopup = () => {
+  const downloadIcons = document.querySelectorAll(".video-download-icons")
+  const popups = document.querySelectorAll(".video-tab-download-popup")
+  let activePopup: HTMLElement | null = null
+
+  if (downloadIcons.length > 0) {
```

Comment:
**Requested Change:** I think you should remove this check: There's no need to check a list's length is nonzero when iterating. Iterating over a zero-length list is totally fine.

_Checking that the list is actually defined, and not `null` or `undefined` might be necessary sometimes. But querySelectorAll **always** returns a list so there's no need for that here_

Reply:
Done